src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1954,17 +1956,38 @@ static Set<String> expandFileNames(String[] filePathList, String queryId)
           result.add(((java.io.File) file).getCanonicalPath());
         }
       } catch (Exception ex) {
-        throw new SnowflakeSQLException(
-            queryId,
-            ex,
-            SqlState.DATA_EXCEPTION,
-            ErrorCode.FAIL_LIST_FILES.getMessageCode(),
-            "Exception: "
-                + ex.getMessage()
-                + ", Dir="
-                + entry.getKey()
-                + ", Patterns="
-                + entry.getValue().toString());
+        if (ex instanceof UncheckedIOException) {
+          if (!dir.exists()) {
+            logger.debug(
+                "The directory: "
+                    + entry.getKey()
+                    + " has been deleted. Ignoring files under this directory.");
+          } else {
+            throw new SnowflakeSQLException(
```

Comment:
let's not create the same exception twice as copy pasted - we can refactor this code to have only one exception thrown when file exist?